### DISCUSSION TITLE
Title: error importing the photoDNA database - IPED 4.1.5

### DISCUSSION THREAD

**Discussion by danpos** [sepinf-inc/IPED#1928] [General]
Good afternoon, I was trying to import the PhotoDNA database into the IPED 4.15 hash base, according to the instructions on the Wiki, but the following error occurred: 

ERROR: File PATH-TO-FILE\FILE.txt header must contain at least one hash column.

As I haven't found any additional information about the error and how to fix it, I'm asking how to proceed. 

Thank you in advance for your attention.

**Comment by wladimirleite** [comment]
Hello @danpos!
The way IPED 4 hash database was designed, there must be at least one hash column (MD5, SHA1, SHA256, SHA512 or EDONKEY).
Below is an example of a PhotoDNA file (MD5 was the used hash).
```
MD5,photoDna,status,set
77AD07C2DA59E42BE3DD868A1429088F,WhY9ONov7S0ppb2WI8WXMw0tQgQiAyUF0Sl0OpTQUv9EmiSvSZMjWBk6IBs4BTgIRGFNRBacDJVGKg6gN0wYey4wFyVaAwwxF/Q1dwOdES8hUg5KXwwIcCkVDkpNCQY3HG4sTxOPUxMDUy8EGAkVGh0LD1oiEQRhEzkwCRZcNRcEaVsDCx1tAwIaWggGEUYU,pedo,LED
14B8602C506A76B4BE04DE8784C920DD,AgERAAcDJwAaBGoANR6uEAFARQgABwMAIwEcALQU6AlTNvwVNpi0QgidTyMADQYAbAMUAu5EOVhdNC5nH4d3OgX3MTAADAcAbAgzBm11K2HHRhGdOq8SkwCRDj8ABBYAZh9ODzB0G0OMNnU0JqREbwAfDgwOAw4AmWl/NCKHGS+FFz2eIaVtjAAnEgdFAg4G,pedo,LED
```

Also note that PhotoDNALookup task use the "status" column.
It matches only PhotoDNAs which status are defined in parameter "statusHashDBFilter" defined in "PhotoDNALookupConfig.txt".
The default configuration is "pedo".

**Reply by danpos** [reply]
Thanks for the reply, @tc-leite.

The version I have was made available to me in 2020 and consists of a file with 2419327 lines containing only the unique PhotoDNA values, with no header and no more columns.

From what I understand from your reply, this "old" version is not compatible with the new HASH database design, so I will have to request the most up-to-date version, in accordance with the rules applicable to such a request.

In any case, thank you once again for your promptness and kindness.

**Reply by wladimirleite** [reply]
Yes, ideally you would have to request another version.
Is it a public database?

A possible (ugly) workaround would be modify you file and add a sequential "fake hash" (e.g. use sha512 and starts with 128 zeroes, incrementing one for each new line).

**Reply by danpos** [reply]
No, I formally requested it from SPINF/DPF through the ICCE-SEDE (RJ) directorate, as I was the head of the forensic computing service (RJ), the file is restricted to law enforcement officials.

I'll try the alternative you suggested.

Thanks again for your reply and guidance.

**Reply by danpos** [reply]
I tried the suggested workaround and got the following error:

[ 0,0% ]
Error reading file: PATH-TO-FILE\FILE.txt 
java.lang.RuntimeException: Record #1: invalid PhotoDna size content:
CSVRecord [comment='null', recordNumber=1, values=[00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000, EDITED]]
at iped.engine.hashdb.HashDBTool.readFile(HashDBTool.java:796)
at iped.engine.hashdb.HashDBTool.readFiles(HashDBTool.java:600)
at iped.engine.hashdb.HashDBTool.run(HashDBTool.java:151)
at iped.engine.hashdb.HashDBTool.main(HashDBTool.java:125)

Any other suggestions?

**Reply by danpos** [reply]
Um, I think I understand the problem. It seems to me that the formatting of the old PhotoDNA hash base is structured as follows:

MD5*PhotoDNA

So there's no need for a workaround, just replace the "*" character with ",".

I'll try this solution and report back.

**Reply by wladimirleite** [reply]
Exactly! It should be enough.
Other than that, you probably will need to set status="pedo" for all lines. It is possible to do that without modifying the CSV file, with a command line parameter.

**Reply by danpos** [reply]
The PhotoDNA database has been imported.
As for the status, according to your first reply, I understood that it wasn't necessary to do this in the file, as the default is already set to 

status=pedo

Now that the import is done, when running iped, should I pass this parameter on the command line? If so, how?

**Reply by wladimirleite** [reply]
The default in PhotoDNALookup is to match only if status ="pedo".
I am not in the PC right now, but if you run the import tool without argument it should show a parameter the allowing setting a property for the whole file.

**Reply by danpos** [reply]
I did what you suggested and the option is as follows:

>  -addCol <column name> <fixed value>
>     Add a fixed column (for all imported lines), with a given name and value.

I'm redoing the import so that I have the database correctly configured for production.

Thanks again for your patience and support.

**Reply by danpos** [reply]
The job was done again and everything went well.
Thanks again for your help.
Question resolved, we can close the discussion.
